main.c

```diff
@@ -16,8 +19,15 @@
 #include "parser.h"
 #include "wlr-output-management-unstable-v1-client-protocol.h"
 
+#ifdef KANSHI_HAS_VARLINK
+#include <varlink.h>
+#include "ipc.h"
+#endif
+
 #define HEADS_MAX 64
 
+#define ARRAY_LENGTH(_arr) (sizeof(_arr) / sizeof(*(_arr)))
```

Comment:
I'd prefer to just inline the `sizeof(arr) / sizeof(arr[0])` expressions

Reply:
Done for event-loop.c